clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -155,6 +163,10 @@ public void addDisconnectionListener(DisconnectionListener listener) {
     disconnectionListeners.add(listener);
   }
 
+  public void addReconnectListener(ReconnectListener listener) {
+    reconnectListeners.add(listener);
```

Comment:
This part where reconnectListener updates reconnect data, (see ReconnectData) can also be refactored in a better way, I can use some suggestions there. IMO, should we have a seperate ReconnectMsg refactored and sent to server